src/commonMain/kotlin/fr/acinq/lightning/io/Peer.kt

```diff
@@ -208,11 +208,16 @@ class Peer(
                 }
         }
         launch {
+            // Wallet utxos that are already used in channel funding attempts should be ignored, otherwise we would double-spend ourselves.
+            // If the electrum server we connect to has our channel funding attempts in their mempool, those utxos wouldn't be added to our wallet anyway.
+            // But we cannot rely only on that, since we may connect to a different electrum server after a restart, or transactions may be evicted from their mempool.
+            // Since we don't have an easy way of asking electrum to check for double-spends, we would end up with channels that are stuck waiting for confirmations.
+            // This generally wouldn't be a security issue (only one of the funding attempts would succeed), unless 0-conf is used and our LSP is malicious.
+            val initiallyReservedUtxos: Set<OutPoint> = Helpers.reservedWalletInputs(db.channels.listLocalChannels())
```

Comment:
We're now reading channels from the DB twice, here and in the coroutine below. Should we group this coroutine and the one below as children of a shared coroutine to list channels only once?

Reply:
Also, we could be disconnected and switch electrum servers while the app stays up, meaning that for best effort we should check channels everytime we are notified of a new wallet state.

A solution could be:
1) isolate the spawning of the swap-in coroutine in a method, say `launchSwapInWallet()`
2) call `launchSwapInWallet()` after `db.channels.listLocalChannels()` in the coroutine that reads the db (it would fit nicely near the "check htlc timeout" coroutine)
3) call `Helpers.reservedWalletInputs(_channels.value)` in the swap-in coroutine

WDYT?

Reply:
I applied 1) and 2) in https://github.com/ACINQ/lightning-kmp/pull/474/commits/88f3c72e4bcc145faccfcb236df7bc3b4512e475

As discussed offline, even if we disconnect from electrum, we don't need to re-scan our channels because we already added the utxos for each `RequestChannelOpen` we made since launch to our `reservedUtxos` set.